### PR TITLE
Feature 2 andriy melnyk

### PR DESCRIPTION
**Pull request by Melnyk9506**
guys pleace chack the code

### CODE REVIEW

File: server/src/main/java/com/lunchtime/controllers/TokenController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/token")
+
+public class TokenController {
+
+    private JwtGenerator jwtGenerator;
+
```
Comment (narekacy) [inline server/src/main/java/com/lunchtime/controllers/TokenController.java:16]: It is a good practice to make constructor fields `final`.

**Comment by VanderbiGit** [comment]
***************************
APPLICATION FAILED TO START
***************************
Description:
Parameter 1 of constructor in com.lunchtime.controllers.UserController required a bean of type 'org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder' that could not be found.
Action:
Consider defining a bean of type 'org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder' in your configuration.
Process finished with exit code 0

Have  you this error?
I can not start project.

**Comment by VanderbiGit** [comment]
You have Checkstyle error.
/home/vadim/Projects/Lunch_Time/server/src/main/java/com/lunchtime/security/JwtAuthenticationProvider.java
Error:(46, 64) java: incompatible types: com.lunchtime.models.Role cannot be converted to java.lang.String

File: server/src/main/java/com/lunchtime/controllers/TokenController.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public TokenController(JwtGenerator jwtGenerator) {
+        this.jwtGenerator = jwtGenerator;
+    }
+
+    @PostMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public String generate(@RequestBody final Person person) {
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/controllers/TokenController.java:22]: This method invoke in login page. In this case object Person doesn't exist. You should use object with `username and password` properties

File: server/src/main/java/com/lunchtime/controllers/TokenController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import com.lunchtime.models.Person;
+import com.lunchtime.security.JwtGenerator;
+import org.springframework.http.MediaType;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/token")
+
+public class TokenController {
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/controllers/TokenController.java:13]: I think better name will be AuthController. And endpoint must start with `/api` - `/api/auth` or `/api/authenticate`

File: server/src/main/java/com/lunchtime/models/Person.java [hunk 4]
```diff
@@ -1,42 +1,42 @@
...
 import java.time.Instant;
+import javax.persistence.*;
+import javax.validation.constraints.Email;
+
+
 
 @Entity
-@Setter
-@Getter
+@Data
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/models/Person.java:14]: Nice!

File: server/src/main/java/com/lunchtime/models/Person.java [hunk 5]
```diff
@@ -1,42 +1,42 @@
...
     @Email
-    @Column(name = "email")
+    @Column(name = "email", unique = true)
     private String email;
+    @NotBlank
+    @Column(name = "phone_nomber")
+    private Integer phone;
 
+    @Column(name = "avatar_url")
+    private String photoUrl;
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/models/Person.java:35]: If you change property name you shouldn't forget to update client part where this property is used

File: server/src/main/java/com/lunchtime/models/Status.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package com.lunchtime.models;
+
+public enum Status {
+    ACTIVE, NOT_ACTIVE, DELETED
```
Comment (vitaliimak) [inline server/src/main/java/com/lunchtime/models/Status.java:4]: We have separate property isDeleted. If you decided to change current behavior with `isDeleted` you should remove `isDeleted` property and setters, getters.
Or you can remove `DELETED` field from this enum and left current behavior

**Review by vitaliimak — CHANGES_REQUESTED** [review]

**Comment by vitaliimak** [comment]
@Melnyk9506 Don't forget to assign PR to yourself

File: server/src/main/java/com/lunchtime/models/Role.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+@Data
+public class Role  {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @Column(name = "name")
+    private String name;
+
```
Comment (Andrii35) [inline server/src/main/java/com/lunchtime/models/Role.java:21]: Should't be advised in this entity relationship to entity Person to?

File: server/src/main/java/com/lunchtime/models/Person.java (hunk 5 above)
Comment (narekacy) [reply server/src/main/java/com/lunchtime/models/Person.java:35]: Question to line 32: Do we need phone number to store in `Integer `or `String`?

**Pull request closed without merging**